test/cli/gen/gen.ml

```diff
@@ -0,0 +1,61 @@
+module StringMap = Map.Make (String)
+
+type entry = {
+  has_ref : bool;
+  has_opts : bool;
+  should_fail : bool
+}
+
+let empty_entry = {
+  has_ref = false;
+  has_opts = false;
+  should_fail = false;
+}
+
+let register_file entries fname =
+  let update_or_add key ~f =
+    StringMap.update key (function
+        | Some entry -> Some (f entry)
+        | None -> Some (f empty_entry)) entries
+  in
+  match String.split_on_char '.' fname with
+  | [ test_name; "ref" ] ->
+      update_or_add test_name ~f:(fun e -> { e with has_ref = true })
+  | [ test_name; "opts" ] ->
+      update_or_add test_name ~f:(fun e -> { e with has_opts = true })
+  | [ test_name; "should-fail" ] ->
+      update_or_add test_name ~f:(fun e -> { e with should_fail = true })
+  | _ ->
+      entries
+
+let check_test test_name entry ok =
+  let e ok b =
+    if b then Format.kfprintf (fun _ -> false) Format.err_formatter
+    else Format.ikfprintf (fun _ -> ok) Format.err_formatter
+  in
+  let ok = e ok (not entry.has_ref) "@{<error>Error@}: Missing file %s.ref\n" test_name in
```

Comment:
I think you can simplify this part by using `invalid_arg` as it's only used to exit with an error message

Reply:
I did that way to print all the warnings before exiting. Do you think I should remove that feature ?